pkg/cpio/sysinfo_freebsd.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2013-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cpio
+
+import (
+	"syscall"
+)
+
+func sysInfo(n string, sys *syscall.Stat_t) Info {
+	return Info{
+		Ino:      sys.Ino,
+		Mode:     uint64(sys.Mode),
+		UID:      uint64(sys.Uid),
+		GID:      uint64(sys.Gid),
+		NLink:    sys.Nlink,
+		FileSize: uint64(sys.Size),
+		Major:    sys.Dev >> 8,
+		Minor:    sys.Dev & 0xff,
```

Comment:
Why does the calculation for major/minor differ from https://github.com/golang/sys/blob/master/unix/dev_freebsd.go ?

Reply:
bigger question: if you are just trying to get u-root to run, this chance should not be needed. This is part of my puzzlement. You are targeting linux, not freebsd, so there are changes here I do not understand.

Reply:
Sorry, I am not familiar yet with the syscalls nor with Go and I just found some of the upstream docs by chance. I commented out parts that kept u-root from being compiled, then gradually copied existing files or, as in this case, hacked old stuff from previous revisions in the repo back in. At some point I had the code compile and I was able to create an initramfs with `GOOS=linux u-root -build=bb` and run it in QEMU with kernels from Arch, DSL and Ubuntu. That made me happy and I wanted to share with upstream, where you know much better how all of this works. :D Thanks for all the feedback! :)